sqlalchemy_redshift/dialect.py

```diff
@@ -631,6 +631,18 @@ def _get_all_relation_info(self, connection, **kw):
     @reflection.cache
     def _get_all_column_info(self, connection, **kw):
         result = connection.execute("""
+        SELECT nspname AS "name"
+        FROM pg_catalog.pg_namespace
+        WHERE nspname !~ '^pg_' AND nspname <> 'information_schema'
+        ORDER BY 1
+        """)
+        schema_names = ['"%s"' % r.name for r in result]
+        search_path = ','.join(schema_names)
+        result = connection.execute("""
+        -- We temporarily set the search_path here to include
+        -- all non-system schemas because pg_table_def doesn't show
+        -- results for schemas not on the search_path.
+        SET LOCAL search_path TO {search_path};
```

Comment:
It's ridiculous that we have to do this, but it works and I think it'll be robust.